os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -137,117 +141,157 @@ static int binary_manager_read_header(int bin_idx, int part_idx, binary_header_t
 	close(fd);
 
 	return OK;
+
 errout_with_fd:
 	close(fd);
-
+	if (need_unlink) {
+		unlink(path);
+	}
 	return ERROR;
 }
 
 /****************************************************************************
- * Name: binary_manager_load_binary
+ * Name: binary_manager_exec_binary
  *
  * Description:
- *	 This function loads binary with index in binary table.
+ *	 This function loads and executes binary.
  *
  ****************************************************************************/
+static int binary_manager_exec_binary(int bin_idx, char *path, binary_header_t *header_data)
```

Comment:
I think this function is not only for execution. It includes load also.
Why did you change the function name?